### PR TITLE
Fix  Open Images uploaded outside of media gallery are not reflected …

### LINKED ISSUE #1428: Images uploaded outside of media gallery are not reflected in the media gallery
### Preconditions (*)
1. Install Magento with Adobe Stock Integration
2. Enable Enhanced Media Gallery in Stores -> Configuration -> System -> Enhanced Media Gallery fieldset

### Steps to reproduce (*)
1. Login to Admin panel
2. Go to Catalog -> Category
3. Upload the Category image using the "Upload button"
4. Save the category
5. Open Media Gallery (i.e. using "Select from Gallery" button for the same Category image field)
6. Go to catalog/category folder in the Media Gallery

### Actual result (*)
The image saved for the category is absent in the media gallery

### Expected result (*)
The image saved for the category should be present in the media gallery

### Technical details
An after plugin on `\Magento\Catalog\Model\ImageUploader::moveFileFromTmp` needs to be introduced to the `Magento_MediaGalleryIntegration` module to save the information about uploaded image using `\Magento\MediaGalleryApi\Api\SaveAssetsInterface`

### LINKED ISSUE #1435: [MFTF] Cover uploading images outside of the Media Gallery
### Preconditions (*)
1. Installed Magento with Adobe Stock Integration
2. Configured  integration in Stores -> Configuration -> System -> Adobe Stock Integration fieldset

### Steps to reproduce (*)
Admin user is logged in to admin panel
Go to Stores - Configuration under Settings section 
Assert that Configuration window is opened
User expands Advanced tab
User click System
User expands Enhanced Media Gallery tab
User selects Yes from Enabled dropdown
User clicks Save config button
Assert that the configuration has been saved 
Navigate to Catalog > Categories
User select Root Category form category tree
User clicks Add Subcategory button
User types Category name
User expands Content
User clicks Upload button
"Open Files" window appears
User selects an image from local storage
User clicks Open button
Assert that Category image is displayed
User Saves category
User expands Content
Assert that the previously uploaded image is displayed in Category Image section
User clicks Select from Gallery button
Media Gallery is opened
Assert that the previously uploaded image exist in Media Gallery 

### Note (*)
Issue - https://github.com/magento/adobe-stock-integration/issues/1428
PR - https://github.com/magento/adobe-stock-integration/pull/1431
Test on Cucumber Studio - https://studio.cucumber.io/projects/131313/test-plan/folders/943908/scenarios/4836631

### PR DESCRIPTION
**Pull request by Nazar65**
…in the media gallery

### Fixed Issues (if relevant)

1. Fixes magento/adobe-stock-integration#1428: Images uploaded outside of media gallery are not reflected in the media gallery
2. Fixes magento/adobe-stock-integration#1435: [MFTF] Cover uploading images outside of the Media Gallery

### Manual testing scenarios (*)
1. Login to Admin panel
2. Go to Catalog -> Category
3. Upload the Category image using the "Upload button"
4. Save the category
5. Open Media Gallery (i.e. using "Select from Gallery" button for the same Category image field)
6. Go to catalog/category folder in the Media Gallery

**The image saved for the category should be present in the media gallery**

### CODE REVIEW

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

File: MediaGalleryIntegration/Plugin/SaveBaseCategoryImageInformation.php [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+ */
+declare(strict_types=1);
+
+namespace Magento\MediaGalleryIntegration\Plugin;
+
+use Magento\MediaGalleryApi\Api\SaveAssetsInterface;
+use Magento\Catalog\Model\ImageUploader;
+use Magento\MediaGallerySynchronization\Model\CreateAssetFromFile;
+use Magento\Cms\Model\Wysiwyg\Images\Storage;
+use Magento\MediaGalleryUi\Model\Filesystem\SplFileInfoFactory;
```
Comment (sivaschenko) [inline MediaGalleryIntegration/Plugin/SaveBaseCategoryImageInformation.php]: Can you please move `SplFileInfoFactory` to `MediaGallerySynchronization` module and remove the dependency on `MediaGalleryUi` module

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Thanks for the pull request @Nazar65 ! Please take a look at my comment.

Can you please also check uploading the image for PageBuilder and probably add a plugin for `\Magento\Framework\File\Uploader::save`

File: MediaGalleryIntegration/Plugin/SaveBaseCategoryImageInformation.php (hunk 1 above)
Comment (Nazar65) [reply MediaGalleryIntegration/Plugin/SaveBaseCategoryImageInformation.php]: ok

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Review by sivaschenko — APPROVED** [review]

**Comment by chalov-anton** [comment]
:heavy_check_mark: QA Passed 
- rechecked on CE instance
- rechecked on EE instance with Page Builder installed

**Pull request merged**